### PR TITLE
many small fixes from running shellcheck

### PR DESCRIPTION
**Pull request by Risca**
shellcheck is a great tool to check your shell code for potential issues. I ran it on all scripts I could find in this repo and fixed most issues that it found. There was no major issues, only minor.

Changelog: None
Signed-off-by: Patrik Dahlström <risca@dalakolonin.se>

### CODE REVIEW

File: tests/run.sh [hunk 1]
```diff
@@ -34,7 +33,7 @@ EOF
 
 function inject_pre_generated_ssh_keys() {
     ssh-keygen -f /tmp/mender-id_rsa -t rsa -N ''
-    echo "cd /home/root/\nmkdir .ssh\ncd .ssh\nwrite /tmp/mender-id_rsa.pub id_rsa.pub\nwrite /tmp/mender-id_rsa id_rsa" | debugfs -w core-image-full-cmdline-vexpress-qemu.ext4
+    printf "cd /home/root/\nmkdir .ssh\ncd .ssh\nwrite /tmp/mender-id_rsa.pub id_rsa.pub\nwrite /tmp/mender-id_rsa id_rsa" | debugfs -w core-image-full-cmdline-vexpress-qemu.ext4
```
Comment (kacf) [inline tests/run.sh]: Missing newline at end, compared to the `echo`.

File: migration/dump-db [hunk 2]
```diff
@@ -1,33 +1,33 @@
...
+  mender-device-adm
+  mender-deployments
+)
+DB_SERVICES=(
+  mender-mongo-useradm
+  mender-mongo-inventory
+  mender-mongo-device-adm
+  mender-mongo-device-auth
+  mender-mongo-deployments
+)
```
Comment (kacf) [inline migration/dump-db]: NACK, let's not use arrays unless we have to.

File: migration/restore-db [hunk 3]
```diff
@@ -1,34 +1,34 @@
...
-  mender-device-adm \
-  mender-deployments \
-"
+SERVICES=(
+  mender-inventory
+  mender-useradm
+  mender-device-auth
+  mender-device-adm
+  mender-deployments
+)
```
Comment (kacf) [inline migration/restore-db]: Same, let's avoid arrays.

**Comment by Risca** [comment]
Updated, but I haven't got the infrastructure setup to test all the changes. Please verify that it works properly before merging :)

File: migration/dump-db [hunk 4]
```diff
@@ -19,15 +19,16 @@ DB_SERVICES="\
...
 
 # dump DB
-DUMP_PATH=$PWD ${SCRIPT} -f ${SELFDIR}/migration-helper.yml \
+DUMP_PATH=$PWD ${SCRIPT} -f "${SELFDIR}/migration-helper.yml" \
          run \
          --rm \
-         -e DB_SERVICES="${DB_SERVICES}" \
          mongo-helper \
-         sh -c 'for s in ${DB_SERVICES}; do mongodump -h "$s" --out "/srv/db-dump/$s"; done'
+         sh -c "for s in ${DB_SERVICES}; do mongodump -h \"\$s\" --out \"/srv/db-dump/\$s\"; done"
```
Comment (kacf) [inline migration/dump-db]: This was originally with single quotes, which looks much better.

File: migration/restore-db [hunk 5]
```diff
@@ -20,15 +20,16 @@ DB_SERVICES="\
...
 
 # dump DB
-DUMP_PATH=$PWD ${SCRIPT} -f ${SELFDIR}/migration-helper.yml \
+DUMP_PATH=$PWD ${SCRIPT} -f "${SELFDIR}/migration-helper.yml" \
          run \
          --rm \
-         -e DB_SERVICES="${DB_SERVICES}" \
          mongo-helper \
-         sh -c 'for s in ${DB_SERVICES}; do mongorestore -h "$s" --drop "/srv/db-dump/$s"; done'
+         sh -c "for s in ${DB_SERVICES}; do mongorestore -h \"\$s\" --drop \"/srv/db-dump/\$s\"; done"
```
Comment (kacf) [inline migration/restore-db]: Same here, the original was better.

**Comment by Risca** [comment]
I'm still a bit new to how GitHub review works so I might have just added a review comment much later than I expected.

Anyway, how do I solve the issue with missing sign-off and changelog?

**Comment by kacf** [comment]
Try `git rebase -i origin/master` (I'm assuming `origin` is your upstream remote). This gives you a list of your commits with `pick` in front of them. Replace `pick` with `reword` for those you want to change, save and exit and you can edit the commit message of each one.

**Comment by Risca** [comment]
I rebased and squashed the fixups into one commit - the later changes are minor anyway. I didn't know `git push -f` was the way to go :open_mouth:

**Comment by kacf** [comment]
Oh yes, absolutely. During reviews it's common to rebase pull requests so they are nice and tidy when it's time to merge.

Looks good now, will wait for the final tests to go through.

**Comment by kacf** [comment]
Thanks @Risca!

**Pull request merged**